### PR TITLE
boards/seeedstudio-gd32: complete and improve board definition

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR completes and improves the board definition for the Seeedstudio GD32 RISC-V Dev board. It
- adds the user LED and Button definitions to `board.h` (bef42091091cfd8d79130f2109b22b2379959057)
- adds the SAUL support for user LED and Button definitions (7a5b2f1fb43f5e28eec4f65c8e7367510be6eb80)
- changes the OpenOCD configuration (32c0c4b1b5b9705ffcd2bb02a1d868ad35ca3bbc)
   - to be usable with Upstream OpenOCD release version 0.12.0
   - to allow different FTDI configuration and other adapters
- adds a documentation with flashing guide, feature list, support status, pinout, schematic reference (9dcc83b8ceb8a9ce526f0a03053a242e866ebf4a)

These changes is the first PR for a number of follow-up PR I will provide in next days to extend the GD32VF103 support. I have already working
- `periph_adc` support,
- `periph_spi` support,
- `tinyusb_device` support,

and almost finished
- `periph_i2c` support (implemented and working with some errors),
- `pm_layered` support (implemented but not working correctly yet), and
- `periph_usbdev` support (implemented by extending `usbdev_synopsys_dwc2 driver but bot working yet).

I will try to implement
- `periph_gpio_irq` support,
- `periph_rtc` support, and
- `periph_rtt` support.

Since I'm using the Sispeed Longan Nano board for testing, I will add the board definition for this board. I will then move some board definitions to a common folder.

### Testing procedure

Green CI.
Documentation should be generated correctly.
Flashing the `seeedstudio-gd32` should still work.
`tests/leds` should work.

### Issues/PRs references

### CODE REVIEW

File: boards/seeedstudio-gd32/doc.txt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+| Family      | RISC-V with ECLIC                      |           |
+| Vendor      | GigaDevice                             |           |
+| RAM         | 32 kByte                               |           |
+| Flash       | 128 KByte                              |           |
+| Frequency   | 108 MHz                                |           |
+| Power Modes | 3 (Sleep, Deep Sleep, Standby)         |   no      |
+| GPIOs       | 80                                     |   yes     |
+| Timers      | 5 x 16-bit timer                       |   yes     |
+| RTC         | 1 x 32-bit counter, 20-bit prescaler   |   no      |
+| Watchog     | 2 x 12-bit counter, 3-bit prescaler    |   no      |
```
Comment (benpicco) [inline boards/seeedstudio-gd32/doc.txt:39]:
```suggestion
| Watchdog    | 2 x 12-bit counter, 3-bit prescaler    |   no      |
```

**Comment by benpicco** [comment]
Please squash

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
Please squash

**Comment by benpicco** [comment]
bors merge

**Comment by kaspar030** [comment]
bors cancel
bors merge

**Pull request merged**